enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+
+#### Summary of the rules outlined above
+
+The following table describes the actions for Machines based on the API to which the action was taken, the action type and whether the Machine has an owner or not.
+
+| Machine Type | Authoritative API | Action | Has Mirror | Has owner | Outcome |
+| --- | --- | --- | --- | --- | --- |
+| Machine API | `machine.openshift.io` | Create | No | - | Sync controller creates Cluster API equivalent and starts synchronisation from Machine API to Cluster API for future updates |
+| Machine API | `machine.openshift.io` | Create | Yes | - | Error - this should be rejected by a webhook |
+| Machine API | `cluster.x-k8s.io`| Create | Yes | - | Sync controller syncs Cluster API spec to Machine API Machine |
+| Machine API | `cluster.x-k8s.io`| Create | No | - | Sync controller creates Cluster API equivalent and starts synchronisation from Cluster API to Machine API for future updates |
+| Machine API | `''` | Create | Yes | - | Annotation is set to `cluster.x-k8s.io` and follows above description |
+| Machine API | `''` | Create | No | - | Annotation is set to `machine.openshift.io` and follows above description |
+| Machine API | - | Delete | Yes | Yes | Sync controller deletes Cluster API equivalent Machine |
+| Machine API | - | Delete | Yes | No | Cluster API equivalent Machine remains in cluster |
+| Cluster API | - | Create | N/A | - | Sync controller takes no action. Cluster API Machine acts independently of Machine API |
+| Cluster API | `machine.openshift.io`  | Create | Yes | - | Sync controller syncs Machine API spec to Cluster API Machine |
+| Cluster API | `cluster.x-k8s.io`  | Create | Yes | - | Sync controller syncs Cluster API spec to Machine API Machine |
+| Cluster API | - | Delete | Yes | Yes | Sync controller deletes Machine API equivalent Machine |
+| Cluster API | - | Delete | Yes | No | Machine API synchronisation controller will recreate Machine after it is removed[^9] |
+
+[^9]: Any Cluster API Machine that is deleted that does not have an owner, but is reflecting a Machine in Machine API,
+is likely to be a control plane Machine without a ControlPlaneMachineSet.
+In this case, to prevent accidental deletion, we do not propagate the delete event.
+This should not interfere with the semantics of any higher level abstraction or resource such as MachineHealthCheck.
+
+#### Filtering reconciles in Cluster API controllers
+
+To enable the Cluster API controllers to observe the authoritative annotation, we will need to implement a check at the beginning of the reconcile logic of each controller.
+
+This will look up the current state of the Machine API equivalent resource and determine whether the controller should be reconciling that specific resource or not.
+This will hopefully be a relatively small carry patch that we will need to carry and should be written in a way that the logic is utilitarian so that we can reduce the duplication of the code.
```

Comment:
I'm wondering if this is something that should be split into a library so that the patches we carry in the controllers is just making calls to the shared logic. I presume this is what you're trying to say here, am I correct?